src/components/Root.tsx

```diff
@@ -157,6 +162,7 @@ export class Root extends React.Component<Props, State> {
     WorkerInstance.onStdOutput = this.onStdOutput_;
     WorkerInstance.onStdError = this.onStdError_;
     WorkerInstance.onStopped = this.onStopped_;
+    this.setState({ loading: false });
```

Comment:
If `loading` is set to `true` in `constructor` and then `false` in `componentDidMount`, will the `<Loading>` ever come into play? I wouldn't expect any delay between `constructor` and `componentDidMount`

Reply:
I tend to see the screen for about half a second, although it could be the `PageRouter` loading screen that I'm seeing, I don't recall if I tested with just this loading screen active. In my performance profile, I saw the Root component taking about 50ms on initial render, which was why I introduced the screen, although I don't think that this screen is necessarily required. `render` is called before `componentDidMount`, and it seemed like there was some amount of logic that could cause `render` to take some time (eg loading the `SimulatorArea`), which was why it was added. I'm fine with removing the screen on this page.

Reply:
Gotcha, just curious. I haven't run it yet, so if it does show up, let's leave it

Reply:
Update after testing: we should likely remove this loading screen, it doesn't come up to the user